supplementary/alica_ros1/supplementary_tests/libalica-supplementary-tests/include/libalica-supplementary-tests/GSolver/GSolverTestPlan.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <boost/dll/alias.hpp>
+#include <engine/BasicCondition.h>
+#include <engine/BasicConstraint.h>
+#include <engine/BasicPlan.h>
+#include <engine/BasicUtilityFunction.h>
+#include <engine/DefaultUtilityFunction.h>
+#include <engine/UtilityFunction.h>
+
+namespace alica
+{
+class GSolverTestPlan : public BasicPlan
+{
+public:
+    GSolverTestPlan(PlanContext& context);
+    virtual ~GSolverTestPlan();
+    static std::unique_ptr<GSolverTestPlan> create(alica::PlanContext& context);
+};
+
+BOOST_DLL_ALIAS(alica::GSolverTestPlan::create, GSolverTestPlan)
+
+class GSolverTestPlanUtilityFunction : public BasicUtilityFunction
+{
+public:
+    std::shared_ptr<UtilityFunction> getUtilityFunction(Plan* plan);
```

Comment:
you can reduce a lot of the repeated code here by using base classes & templates as done in lbc. Since you have already done the changes though, its okay, but you can use this method in other PRs